apps/arweave/src/ar_tx_queue.erl

```diff
@@ -238,35 +236,17 @@ maybe_drop(Q, Size, MaxSize) ->
 maybe_drop(Q, Size, MaxSize, DroppedTXs) ->
 	case Size > MaxSize of
 		true ->
-			{NewQ, TX, DroppedSize} = drop_lowest(Q),
+			{{_, {TX, DroppedSize}}, NewQ} = gb_sets:take_smallest(Q),
 			maybe_drop(NewQ, Size - DroppedSize, MaxSize, [TX | DroppedTXs]);
 		false ->
 			{Q, Size, lists:filter(fun(TX) -> TX /= none end, DroppedTXs)}
 	end.
 
-%% @doc Drop lowest scoring item(s) from priority queue.
-%% NOTE relies on the internal implementation of priority_queue.
-drop_lowest(Q = {pqueue, []}) ->
-	{Q, none, 0};
-drop_lowest({pqueue, Qs}) ->
-	[{LP, LQ} | Rest] = lists:reverse(Qs),
-	case priority_queue:out(LQ) of
-		{empty, _} ->
-			{{pqueue, lists:reverse(Rest)}, none, 0};
-		{{value, {TX, Size}}, QO} ->
-			ar:info([
-				{event, dropped_tx},
-				{id, ar_util:encode(TX#tx.id)},
-				{reason, lowest_utility_in_large_queue}
-			]),
-			{{pqueue, lists:reverse([{LP, QO} | Rest])}, TX, Size}
-	end;
-drop_lowest(X) ->
-	{X, none, 0}.
-
 show_queue(Q) ->
-	[{ar_util:encode(TX#tx.id), TX#tx.reward, Size}
-	 || {_, {TX, Size}} <- priority_queue:to_list(Q)].
+	lists:reverse([
```

Comment:
I don't know if we call `show_queue` often but if we wanted to avoid `reverse`ing the list, we could use `foldl`:
```erlang
show_queue(Q) ->
    lists:foldl(fun({_, {TX, Size}}, Acc) ->
        [{ar_util:encode(TX#tx.id), TX#tx.reward, Size} | Acc]
    end, [], gb_sets:to_list(Q)).
```
But if show_queue isn't called so often, I actually find the reverse more readable.